x-pack/plugins/security_solution/public/common/components/event_details/insights/related_alerts_by_session.tsx

```diff
@@ -79,13 +82,31 @@ export const RelatedAlertsBySession = React.memo<Props>(
       } else if (isEmpty && state !== 'loading') {
         return SESSION_EMPTY;
       }
+      const ensureOnlyRuleProviders: DataProvider = {
+        enabled: true,
+        id: ALERT_RULE_UUID,
+        name: ALERT_RULE_UUID,
+        excluded: false,
+        kqlQuery: '',
+        queryMatch: {
+          field: ALERT_RULE_UUID,
+          value: '*',
+          operator: EXISTS_OPERATOR,
+        },
+      };
+      const dataProvidersWithAlertFilter = cellData.dataProviders.map((provider) => {
```

Comment:
If `cellData` already comes from `useActionCellDataProvider`, the contents of `ensureOnlyRuleProviders` should already be present here and we would not need to add them again. Or am I missing something?